### PR TITLE
Add AoE Flamethrower

### PR DESCRIPTION
**Pull request by dfuchs2**
Added a second flamethrower with AoE

### CODE REVIEW

**Comment by dfuchs2** [comment]
As griped in https://github.com/bleehu/Compound_X/issues/599

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt [hunk 1]
```diff
@@ -94,6 +94,32 @@ Movement Speed Penalty: -2m
...
+Mag Size:  8 units of 6 liters, Naphitic Gel. $320 per magazine
+Reload DC:  50
+Fire Modes:  Semi
+Armor Piercing Level: n/a
+Reflex Modifier: -20
+Suppression Level: Medium
+Movement Speed Penalty: -2m
+
+	This flame thrower fires a wide spread of flaming gel, hitting the target 
+space as well as each adjacent space visible to the wielder.
```
Comment (bleehu) [inline Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: For clarity, can you catch the square on fire, but miss the target if you don't roll accuracy high enough? What happens to a character that enters, leaves, moves through or ends their turn in a square that is burning? How long does the burning last?

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt [hunk 2]
```diff
@@ -94,6 +94,32 @@ Movement Speed Penalty: -2m
 * 4 out to 45m
 * 5 out to 80m
 
+==Frontier Flame Thrower FLMR-B $3880 ==
```
Comment (bleehu) [inline Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: That's a steep price hike over the level 1 flamethrower. So long as it's worth what you're paying for that's fine.

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt [hunk 3]
```diff
@@ -94,6 +94,32 @@ Movement Speed Penalty: -2m
 * 4 out to 45m
 * 5 out to 80m
 
+==Frontier Flame Thrower FLMR-B $3880 ==
+
+Damage: 80 Special
```
Comment (bleehu) [inline Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: 20 less damage than the level 1 flame thrower? I suppose there's the potential to deal it to multiple targets, so maybe that makes sense. Can we change it to deal fire damage? https://github.com/bleehu/Compound_X/blob/play/01_04_Damage_Types_and_Armor_Types.txt#L51

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt [hunk 4]
```diff
@@ -94,6 +94,32 @@ Movement Speed Penalty: -2m
 * 4 out to 45m
 * 5 out to 80m
 
+==Frontier Flame Thrower FLMR-B $3880 ==
+
+Damage: 80 Special
+Mag Size:  8 units of 6 liters, Naphitic Gel. $320 per magazine
```
Comment (bleehu) [inline Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: I like the relatively low ammo capacity. This thing hits like a truck. Slowing the user down to reload often I think creates a beautiful cadence for combat.

**Review by bleehu — CHANGES_REQUESTED** [review]
Blocker: flamethrower added in the machine gun section.

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt (hunk 1 above)
Comment (dfuchs2) [reply Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: I hadn't considered that. I imagine the flame would burn on the ground for one additional turn. I was imagining it as a faster-burning napalm. I'm thinking of making it such that the player rolls dex for each burning space they step in, including the one they were in when the flame thrower hit. Anybody who ends their turn in the fire would roll dex again one more time before the fire diminishes. The burning damage from the fire on the ground would be in the range of 30-40.

**Comment by dfuchs2** [comment]
I can't believe that I put it in the wrong section. That was definitely not intentional.

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt (hunk 3 above)
Comment (dfuchs2) [reply Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: I see the original flame thrower is special damage. Should that also be fire damage?

**Comment by dfuchs2** [comment]
As a side note, I was wondering about using flamethrowers in a vacuum.  In reality, flamethrowers were commonly used to deoxegenate bunkers and caves. Since there is no oxygen in space, do our flamethrowers work there? Do they have their own oxygen supplies?

**Comment by bleehu** [comment]
That is a very good question! There's a couple of different directions you could take this. 
1) fire needs oxygen. No oxygen, no flamethrower. Tactical puzzle time!
2) flamethrower propellant is made of stuff that has its own oxygen chemically suspended in the gel like napalm does. It burns in space, under water, and that's pretty much all there is to it. Keeps the mechanics nice and simple.
3) fire burns more quickly with more oxygen, maybe using the flamethrower in an airless environment only reduces the flame counters it gives and maybe it reduces the damage?

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt [hunk 5]
```diff
@@ -320,6 +294,37 @@ listed value when used in this gun. 2 action reload time.
...
+Reload DC:  50
+Fire Modes:  Semi
+Armor Piercing Level: n/a
+Reflex Modifier: -20
+Suppression Level: Medium
+Movement Speed Penalty: -2m
+
+	This flame thrower fires a wide spread of flaming gel, hitting the target 
+space as well as each adjacent space visible to the wielder. The flaming gel 
+continues to burn for one round of turns.
```
Comment (bleehu) [inline Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt:310]: Very specific! I like it!

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt [hunk 6]
```diff
@@ -320,6 +294,37 @@ listed value when used in this gun. 2 action reload time.
...
+	Catches flammable objects on fire. Not affected by armor, unless target is 
+100% sealed. Any targets that took damage apply 1 x Burning @ DC70% to remove. 
+Apply 1 x Fear on hit. No optics or barrel attachments.
+	When the flame hits its target space, anybody who takes a step into a 
+space that has flame on it or begins their turn in one must make a DC60 DEX roll 
+to avoid taking damage from lingering flame on the ground. Failing this roll 
+deals 30 damage and 1 x Burning @ DC70% to remove.
```
Comment (bleehu) [inline Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt:317]: simple succinct and I don't think it can be misinterpreted!

**Review by bleehu — APPROVED** [review]
It seems aesthetically good to me, I like the mechanics. Shipit!

File: Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt (hunk 2 above)
Comment (Turtlelord26) [reply Items/04_06_MachineGuns_HeavyWeps_&_RocketLaunchers.txt]: We can punch the stats into 1source's spreadsheet and see what it tells us 🐢

**Comment by Turtlelord26** [comment]
Notes to self: rules for aoe and relevant damage types in nonstandard gravitatational/atmospheric conditions. We shouldn't worry about those in individual weapon descriptions; they should be more general.

**Comment by Turtlelord26** [comment]
I kinda like the idea that blee's number one is a default but there's a specialty ammo to make number two happen.

**Comment by dfuchs2** [comment]
Perhaps we should stick with number one, and then add number two as a fuel/oxidizer mix? We could increase price by 1.5x and reduce ammo per magazine by 1/3? (Due to the added mass of our oxygen)

**Comment by Turtlelord26** [comment]
New issue outside this immediate scope

**Comment by dfuchs2** [comment]
True. We can add that in as a new ammo type, but this weapon is probably done as is.

**Comment by bleehu** [comment]
Than hit that big green merge button Shrub! You earned it! :)

**Pull request merged**